### PR TITLE
Function not

### PR DESCRIPTION
**Pull request by SadiHassan**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)
- 
### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

not function returning NULL for string input.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Functions/Kusto/kqlNot.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    return DataTypeFactory::instance().get("Bool");
+}
+
+ColumnPtr FunctionKqlNot::executeImpl(
+    const ColumnsWithTypeAndName & arguments, const DataTypePtr & /*result_type*/, const size_t input_rows_count) const
+{
+    const auto & elem = arguments[0];
+
+    if (input_rows_count == 0)
+        return arguments[0].column;
```
Comment (kashwy) [inline src/Functions/Kusto/kqlNot.cpp:61]: seems this is not necessary, and if `input_rows_count == 0 `  is `arguments[0]`  valid ?

File: tests/queries/0_stateless/02366_kql_func_general.sql [hunk 2]
```diff
@@ -86,7 +85,8 @@ Customers | limit toscalar(Customers | where Age > toscalar(toscalar(print 5, 'a
 
 print '-- not --';
 print t = not(1);
-print t = not(false);
```
Comment (kashwy) [inline tests/queries/0_stateless/02366_kql_func_general.sql:89]: need such cases:
print t = not(false);
print t = not(true);

File: src/Functions/Kusto/kqlNot.cpp (hunk 1 above)
Comment (SadiHassan) [reply src/Functions/Kusto/kqlNot.cpp]: This part was added to test when the function was not working, removed now.

File: tests/queries/0_stateless/02366_kql_func_general.sql (hunk 2 above)
Comment (SadiHassan) [reply tests/queries/0_stateless/02366_kql_func_general.sql:89]: Added.

**Review by kashwy — APPROVED** [review]

**Pull request merged**